seed/xylem/gluster.py

```diff
@@ -144,17 +148,23 @@ def call_createdirs(self, args):
     def call_createvolume(self, args):
 
         name = args['name']
+        vol = yield self.getVolume(name)
 
-        vols = yield self.getVolumes()
+        if vol is None:
+            # The volume doesn't exist, let's create it.
+            yield self.createVolume(name)
+            vols = yield self.getVolumes()
```

Comment:
I have not heard that explanation for something like this before.. but trust your judgement.